controllers/metal3machinetemplate_controller_test.go

```diff
@@ -70,7 +73,98 @@ var _ = Describe("Metal3MachineTemplate controller", func() {
 			Namespace: namespaceName,
 		},
 	}
+	type TestCaseM3MoM3MT struct {
+		M3Machine     *capm3.Metal3Machine
+		M3MTemplate   *capm3.Metal3MachineTemplate
+		ExpectRequest bool
+	}
+	DescribeTable("Cluster To Metal3Machines tests",
+		func(tc TestCaseM3MoM3MT) {
+			r := Metal3MachineTemplateReconciler{}
+			obj := client.Object(tc.M3Machine)
+			reqs := r.Metal3MachinesToMetal3MachineTemplate(obj)
 
+			if tc.ExpectRequest {
+				Expect(len(reqs)).To(Equal(1), "Expected 1 request, found %d", len(reqs))
+				Expect(tc.M3Machine.Annotations[clonedFromName]).To(Equal(tc.M3MTemplate.Name))
+				Expect(tc.M3Machine.Annotations[clonedFromGroupKind]).To(Equal("Metal3MachineTemplate.infrastructure.cluster.x-k8s.io"))
+				Expect(tc.M3Machine.Namespace).To(Equal(tc.M3MTemplate.Namespace))
+			} else {
+				Expect(len(reqs)).To(Equal(0), "Expected 0 request, found %d", len(reqs))
+			}
+		},
+		Entry("Reconciliation should not be requested due to missing reference to a template",
+			TestCaseM3MoM3MT{
+				M3Machine: &capm3.Metal3Machine{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "machine-1",
+						Namespace: "bar",
+						Annotations: map[string]string{
+							baremetal.HostAnnotation: "myns/myhost",
+						},
+					},
+					Spec: capm3.Metal3MachineSpec{
+						AutomatedCleaningMode: utils.StringPtr("disabled"),
+					},
+				},
+				M3MTemplate: &capm3.Metal3MachineTemplate{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: capm3.GroupVersion.String(),
+						Kind:       "Metal3MachineTemplate",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "foo",
+					},
+					Spec: capm3.Metal3MachineTemplateSpec{
+						Template: capm3.Metal3MachineTemplateResource{
+							Spec: capm3.Metal3MachineSpec{
+								AutomatedCleaningMode: utils.StringPtr("disabled"),
+							},
+						},
+					},
+				},
+				ExpectRequest: false,
+			},
+		),
+		Entry("Reconciliation should be requested",
+			TestCaseM3MoM3MT{
+				M3Machine: &capm3.Metal3Machine{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "machine-1",
+						Namespace: "foo",
+						Annotations: map[string]string{
+							"cluster.x-k8s.io/cloned-from-name":      "abc",
+							"cluster.x-k8s.io/cloned-from-groupkind": "Metal3MachineTemplate.infrastructure.cluster.x-k8s.io",
+						},
+					},
+					Spec: capm3.Metal3MachineSpec{
+						AutomatedCleaningMode: utils.StringPtr("disabled"),
+					},
+				},
+				M3MTemplate: &capm3.Metal3MachineTemplate{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: capm3.GroupVersion.String(),
+						Kind:       "Metal3MachineTemplate",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "abc",
+						Namespace: "foo",
```

Comment:
I think it would be nice to have variables for the name and namespace. It would make it more obvious that they must be the same as in the annotation.